app/src/commonMain/kotlin/io/github/cgr/app/Assembly.kt

```diff
@@ -0,0 +1,105 @@
+package io.github.cgr.app
+
+import io.github.cgr.app.io.MAGIC_NUMBER
+import io.github.cgr.app.io.Packer
+import io.github.cgr.app.objects.BaseLabel
+import io.github.cgr.app.objects.Opcode
+
+fun assemble(program: String): ByteArray {
+    registerBase()
+    val opcodes = program.lines()
+        .asSequence()
+        .map(String::trim)
+        .filter(String::isNotEmpty)
+        .filter { !it.startsWith('#') }
+        .map(::parseLine)
+        .toList()
+    val constants = opcodes.flatMap { it.second }
+        .filter { (it.startsWith('"') && it.endsWith('"')) || (it.startsWith('[') && it.endsWith(']')) }
```

Comment:
JS brain says regex time

Reply:
Sadly, the regex could match `[dhiag"` (no recursion in JVM regexes), so I can't do that here